src/API/useFetch.tsx

```diff
@@ -1,20 +1,27 @@
 //@ts-nocheck
-import React, { useEffect, useState } from "react";
+import { useEffect, useState } from "react";
 import axios from "axios";
 const GITHUB_BASE_URL = 'https://api.github.com';
 
+export interface repositoryCommitMessage  {
+  date: {}
+  commitMessage: "string"
+  url: "string",
+  username: "string"
+} 
+interface repositoryCommitMessages extends Array<repositoryCommitMessage>{}
 
 export const repositoryFetch = async( queryText: string)  => {
   const data = await axios.get(`${GITHUB_BASE_URL}/search/repositories?q=${encodeURIComponent(queryText)}`);
   const items = data.data.items
   const repositoryChoices = shapeData(items, normalizeRepoData)
-  console.log({ repositoryChoices })
   return repositoryChoices
 };
 
 
-export const useFetch = (repositoryOwner: string, repositoryName: string) => {
-  const [data, setData] = useState([])
+
+export const useFetch = (repositoryOwner: string, repositoryName: string, fire: boolean) => {
+  const [data, setData ] = useState([])
```

Comment:
This is fancy good job. You can make it fancier this way.  Also remove the comment at the bottom of the page.
https://kentcdodds.com/blog/stop-using-isloading-booleans